main.tf

```diff
@@ -1,3 +1,44 @@
-/********************************************************************
-This file is used to implement the ROOT module.
-*********************************************************************/
+##############################################################################
+# Watsonx Assistant
+##############################################################################
+
+locals {
+  watsonx_assistant_crn           = var.existing_watsonx_assistant_instance_crn != null ? data.ibm_resource_instance.existing_assistant_instance[0].crn : ibm_resource_instance.watsonx_assistant_instance[0].crn
+  watsonx_assistant_guid          = var.existing_watsonx_assistant_instance_crn != null ? data.ibm_resource_instance.existing_assistant_instance[0].guid : ibm_resource_instance.watsonx_assistant_instance[0].guid
+  watsonx_assistant_name          = var.existing_watsonx_assistant_instance_crn != null ? data.ibm_resource_instance.existing_assistant_instance[0].resource_name : ibm_resource_instance.watsonx_assistant_instance[0].resource_name
+  watsonx_assistant_plan_id       = var.existing_watsonx_assistant_instance_crn != null ? null : ibm_resource_instance.watsonx_assistant_instance[0].resource_plan_id
+  watsonx_assistant_dashboard_url = var.existing_watsonx_assistant_instance_crn != null ? null : ibm_resource_instance.watsonx_assistant_instance[0].dashboard_url
+}
+
+data "ibm_resource_instance" "existing_assistant_instance" {
+  count      = var.existing_watsonx_assistant_instance_crn != null ? 1 : 0
+  identifier = var.existing_watsonx_assistant_instance_crn
+}
+
+resource "ibm_resource_instance" "watsonx_assistant_instance" {
+  count             = var.existing_watsonx_assistant_instance_crn != null ? 0 : 1
+  resource_group_id = var.resource_group_id
+  name              = var.existing_watsonx_assistant_instance_name != null ? var.existing_watsonx_assistant_instance_name : var.watsonx_assistant_name
+  location          = var.region
+  service           = "conversation"
+  plan              = var.watsonx_assistant_plan
+  tags              = var.resource_tags
+  service_endpoints = var.watsonx_assistant_service_endpoints
+
+  timeouts {
+    create = "15m"
+    update = "15m"
+    delete = "15m"
+  }
+}
+
+##############################################################################
+# Attach Access Tags
+##############################################################################
+
+resource "ibm_resource_tag" "watsonx_assistant_tag" {
+  count       = length(var.access_tags) == 0 ? 0 : 1
+  resource_id = var.existing_watsonx_assistant_instance_crn != null ? var.existing_watsonx_assistant_instance_crn : ibm_resource_instance.watsonx_assistant_instance[0].crn
```

Comment:
why are you not using `local.watsonx_assistant_crn` here?

Reply:
Oh yes, This I missed checking the locals block. Thank you for mentioning.